### PR TITLE
[WFLY-16296][WFLY-17033][WFLY-17034][WFLY-17035] EE namespace interoperability javax.* <-> jakarta.*

### PR DESCRIPTION
**Pull request by fl4via**
Jiras:
 [WFLY-16296](https://issues.redhat.com/browse/WFLY-16296): interoperability work, requires upgrade of WildFly HTTP Client, WildFly Naming Client and JBoss EJB Client
 [WFLY-17033](https://issues.redhat.com/browse/WFLY-17033): upgrade WildFly Naming Client
 [WFLY-17034](https://issues.redhat.com/browse/WFLY-17034): upgrade WildFly HTTP Client
 [WFLY-17035](https://issues.redhat.com/browse/WFLY-17035): upgrade JBoss EJB Client

### CODE REVIEW

**Comment by fl4via** [comment]
@bstansberry tbh, I'm not sure if these PRs should be on a WFLY-16296. Maybe it makes more sense to have the wildlfy-http-common one at the same PR that upgrades wildfly-http-client, and have another Jira filed for adding wildfly-naming-client-jakarta (this one won´t  contain an upgrade, afacs)

**Comment by bstansberry** [comment]
@fl4via I think this is fine, but if you prefer other JIRAs that is fine too.

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/wildfly/http-client/common.main/module.xml [hunk 1]
```diff
@@ -19,24 +19,24 @@
...
-  ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+  ~ 2110-1301 USA, or see the FSF site: http://www.fsf.org.
   -->
 
-<module xmlns="urn:jboss:module:1.9" name="org.wildfly.naming-client">
+<module xmlns="urn:jboss:module:1.9" name="org.wildfly.http-client.common">
 
     <resources>
-        <artifact name="${org.wildfly:wildfly-naming-client}"/>
+        <artifact name="\${org.wildfly.wildfly-http-client:wildfly-http-client-common@module.jakarta.suffix@}"/>
```
Comment (bstansberry) [inline ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/wildfly/http-client/common.main/module.xml:28]: You can just make this:

${org.wildfly:wildfly-naming-client-jakarta}

The complex string replacement syntax let us produce both an EE 8 and an EE 9+ server with the same module.xml file, but we don't need to do that any more.

File: servlet-feature-pack/common/src/main/resources/modules/system/layers/base/org/wildfly/http-client/common/main/module.xml [hunk 2]
```diff
@@ -25,7 +25,7 @@
 <module xmlns="urn:jboss:module:1.9" name="org.wildfly.http-client.common">
 
     <resources>
-        <artifact name="${org.wildfly.wildfly-http-client:wildfly-http-client-common}"/>
+         <artifact name="\${org.wildfly.wildfly-http-client:wildfly-http-client-common@module.jakarta.suffix@}"/>
```
Comment (bstansberry) [inline servlet-feature-pack/common/src/main/resources/modules/system/layers/base/org/wildfly/http-client/common/main/module.xml:28]: You can just make this:

${org.wildfly.wildfly-http-client:wildfly-http-client-common-jakarta}

**Review by bstansberry — CHANGES_REQUESTED** [review]
Hi @fl4via 

Besides the minor module.xml comments I made -- where what you have is ok but would eventually be changed to the simpler form -- the rest looks good, except the licenses.xml files need updating:

https://github.com/wildfly/wildfly/blob/main/ee-feature-pack/common/src/main/resources/license/ee-feature-pack-common-licenses.xml

https://github.com/wildfly/wildfly/blob/main/servlet-feature-pack/common/src/main/resources/license/servlet-feature-pack-common-licenses.xml

The entries in those files should align with the dependencies in the same module's pom.

**Comment by fl4via** [comment]
> @fl4via jboss-ejb-client 4.0.46.Final is available now. If you want to include its upgrade in this PR, that's fine. [WFLY-17035](https://issues.redhat.com/browse/WFLY-17035) Upgrade jboss-ejb-client from 4.0.45.Final to 4.0.46.Final

@chengfang thanks I will do so

File: ee-9/source-transform/naming/pom.xml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        <artifactId>wildfly-ee-9-source-transform-parent</artifactId>
+        <!--
+        Maintain separation between the artifact id and the version to help prevent
+        merge conflicts between commits changing the GA and those changing the V.
+        -->
+        <version>27.0.0.Beta1-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>wildfly-naming-jakarta</artifactId>
```
Comment (bstansberry) [inline ee-9/source-transform/naming/pom.xml:38]: @fl4via I think this file is cruft. It's not wired into the overall build (i.e. no module element in ee-9/source-transform/pom.xml) and the artifact isn't used anywhere.

Comment (fl4via) [reply ee-9/source-transform/naming/pom.xml:38]: ok, I'll delete it @bstansberry

**Review by bstansberry — APPROVED** [review]
@fl4via @chengfang @ropalka These changes look good and pass CI, so I'm merging to get routine testing, including TCK. But please look into https://issues.redhat.com/browse/WFLY-16646 which includes one form of test coverage for this.

**Pull request merged**